salt/orchestrate/services/elasticsearch.sls

```diff
@@ -1,14 +1,17 @@
 {% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
 {% set ENVIRONMENT = salt.environ.get('ENVIRONMENT', 'rc-apps') %}
 {% set env_data = env_settings.environments[ENVIRONMENT] %}
+{% set purpose_data = env_data.purposes.elasticsearch|default({}) %}
 {% set VPC_NAME = env_data.vpc_name %}
-{% set INSTANCE_COUNT = salt.environ.get('INSTANCE_COUNT', 3) %}
+{% set INSTANCE_COUNT = salt.environ.get('INSTANCE_COUNT', purpose_data.num_instances|default(3)) %}
 {% set BUSINESS_UNIT = salt.environ.get('BUSINESS_UNIT', env_data.business_unit) %}
 {% set launch_date = salt.status.time(format="%Y-%m-%d") %}
 {% set subnet_ids = salt.boto_vpc.describe_subnets(
     vpc_id=salt.boto_vpc.describe_vpcs(
         name=env_data.vpc_name).vpcs[0].id
     ).subnets|map(attribute='id')|list %}
+{% set service_name = "{{ ENVIRONMENT }}-elasticsearch" %}
```

Comment:
@blarghmatey Is this going to have any negative effect? I believe that the `profile_name` in https://github.com/mitodl/salt-ops/blob/746e137e32b665c860fcb1778c1e5dcb270ffc63/salt/orchestrate/aws/map_templates/instance_map.yml#L6 is not going to match the top-level property in https://github.com/mitodl/salt-ops/blob/746e137e32b665c860fcb1778c1e5dcb270ffc63/salt/orchestrate/aws/cloud_profiles/elasticsearch.conf#L2

Reply:
Correct, that won't match. For this PR just go ahead and get rid of the service_name override and just restrict the changes to making the instance size, disk size, etc pull from environment settings for generating the cloud map file. Don't change the map template or the profile config.

Reply:
Ok! I've pushed https://github.com/mitodl/salt-ops/pull/782/commits/a432876c6cba1cfe701498955cbfb103ac748750

`instance_map.yml` and `elasticsearch.conf` now have no differences vs `master`.